File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 7/12/2026, 6:25:46 PM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        pref = strs[0]
        res = ""

        for i in range(len(pref)):
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return res
                
            res += s[i]

        return res
        

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 7/12/2026, 6:25:49 PM
class Solution {
    public String longestCommonPrefix(String[] strs) {

        if(strs.length == 0) return "";

        StringBuilder str = new StringBuilder(100);

        for(int i=0; i < strs[0].length() ; i++){

            for(int j = 1; j < strs.length; j++){
                if( strs[j].length() == i || strs[j].charAt(i) != strs[0].charAt(i)){
                    return str.toString();
                }
            }

            str.append(strs[0].charAt(i));
        }

        return str.toString();
    }
}